### PR TITLE
feat: send Person record from full trainee request

### PR DESCRIPTION
**Pull request by Judge40**
When handling a full trainee data request the Person record should be
included and sent back to Kinesis for processing.

TIS21-2997
TIS21-3074

### CODE REVIEW

**Comment by Judge40** [comment]
Created as draft due to dependency on #176

**Review by JosephKelly — DISMISSED** [review]
Looks good so far

File: src/main/java/uk/nhs/tis/sync/dto/PersonDmsDto.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+/**
+ * A DTO representing the expected structure for Person via DMS.
+ */
+@Data
+public class PersonDmsDto {
+
+  private final String id;
+  private final String intrepidId;
+  private final String addedDate;
+  private final String amendedDate;
```
Comment (ReubenRobertsHEE) [inline src/main/java/uk/nhs/tis/sync/dto/PersonDmsDto.java:14]: Is there a reason we include amendedDate here, but not for e.g. PlacementSpecialty? I don't think we use it anywhere downstream (it doesn't make it into the TraineeProfile) - which is not to say its not useful, but I'm wondering if we need to include this field in any of the other DmsDtos?

**Review by JosephKelly — APPROVED** [review]
Reuben's question is a good one, that has been briefly discussed without conclusion.  I'm happy with whichever way that discussion goes.  I don't think either creates more of a mess so reviewing to say as much.

File: src/main/java/uk/nhs/tis/sync/dto/PersonDmsDto.java (hunk 1 above)
Comment (Judge40) [reply src/main/java/uk/nhs/tis/sync/dto/PersonDmsDto.java:14]: The fields match those in the database table, while we don't use most of the fields I wanted to imitate the data we get from DMS as closely as possible.
There's definitely an argument for trimming these DTOs right back to only what TSS needs, but I don't think we should do that until (if) we pull all of this in to it's own data request service

Comment (Judge40) [reply src/main/java/uk/nhs/tis/sync/dto/PersonDmsDto.java:14]: As for why there are fields missing from PlacementSpecialty, it's either an oversight or they weren't available. I can't see a valid reason for it, but I didn't have much involvement in that initial work.

Comment (ReubenRobertsHEE) [reply src/main/java/uk/nhs/tis/sync/dto/PersonDmsDto.java:14]: Ah, it was that Marcello cowboy! :trollface: 

I agree with keeping the fields as similar to the data from DMS as possible at this point. I guess that Reuben cowboy should add amendedDate to the CurriculumMembership and Qualification DmsDtos now 💥

**Pull request merged**